src/app-components/table/Table.tsx

```diff
@@ -0,0 +1,161 @@
+import cn from 'classnames';
+import { format, isValid, parseISO } from 'date-fns';
+import { pick } from 'dot-object';
+
+import classes from 'src/app-components/table/Table.module.css';
+
+interface Column {
+  /** Header text for the column */
+  header: React.ReactNode;
+  /** Keys of the data item to display in this column */
+  accessors: string[];
+  /** Optional function to render custom cell content */
+  renderCell?: (values: string[], rowData: object) => React.ReactNode;
+}
+
+export interface TableActionButton {
+  onClick: (rowIdx: number, rowData: object) => void;
+  buttonText: React.ReactNode;
+  icon: React.ReactNode;
+  color?: 'first' | 'second' | 'success' | 'danger' | undefined;
+  variant?: 'tertiary' | 'primary' | 'secondary' | undefined;
+}
+
+interface DataTableProps<T> {
+  /** Array of data objects to display */
+  data: T[];
+  /** Configuration for table columns */
+  columns: Column[];
+  caption?: React.ReactNode;
+  /** Optional configuration for action buttons */
+  actionButtons?: TableActionButton[];
+  /** Accessible header value for action buttons for screenreaders */
+  actionButtonHeader?: React.ReactNode;
+  /** Displays table in mobile mode */
+  mobile?: boolean;
+  size?: 'sm' | 'md' | 'lg';
+  zebra?: boolean;
+}
+
+function formatIfDate(value: unknown): string {
+  if (typeof value === 'string') {
+    const parsedDate = parseISO(value);
+    if (isValid(parsedDate)) {
+      return format(parsedDate, 'dd.MM.yyyy');
+    }
+  }
+  return String(value);
+}
+
+export function AppTable<T extends object>({
+  caption,
+  data,
+  columns,
+  actionButtons,
+  mobile,
+  actionButtonHeader,
+  size,
+  zebra,
+}: DataTableProps<T>) {
+  const defaultButtonVariant = mobile ? 'secondary' : 'tertiary';
+  return (
+    <Table
+      size={size || 'sm'}
+      className={cn(classes.table, { [classes.mobileTable]: mobile })}
+      zebra={zebra}
+    >
+      {caption}
+      <Table.Head>
+        <Table.Row>
+          {columns.map((col, index) => (
+            <Table.HeaderCell key={index}>{col.header}</Table.HeaderCell>
+          ))}
+
+          {actionButtons && actionButtons.length > 0 && (
+            <Table.HeaderCell>
+              <span className={classes.visuallyHidden}>{actionButtonHeader}</span>
+            </Table.HeaderCell>
+          )}
+        </Table.Row>
+      </Table.Head>
+      <Table.Body>
+        {data.map((rowData, rowIndex) => (
+          <Table.Row key={rowIndex}>
+            {columns.map((col, colIndex) => {
+              const cellValues = col.accessors
+                .filter((accessor) => !!pick(accessor, rowData))
+                .map((accessor) => pick(accessor, rowData));
+              if (cellValues.every((value) => value == null)) {
+                return (
+                  <Table.Cell
+                    key={colIndex}
+                    data-header-title={col.header}
+                  >
+                    -
```

Comment:
Why do we need to show a hyphen for empty cells? No other table components do that as I'm aware of. If it's a matter of taste, it should probably be configurable.